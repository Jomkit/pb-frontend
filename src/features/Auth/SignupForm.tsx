import * as Yup from 'yup';
import { IUser } from '../../types';
import DynamicForm from '../../components/DynamicForm';

const signupSchema = Yup.object().shape({
    username: Yup.string().required(),
    password: Yup.string().min(6).required(),
    firstName: Yup.string().notRequired(),
    lastName: Yup.string().notRequired(),
    email: Yup.string().email().notRequired()
})

const SignupForm = ({signup}: {signup: (values: IUser) => void}) => {
    const initialUserValues: IUser = {
        username: '',
        password: '',
        firstName: '',
        lastName: '',
        email: ''
    }
    const handleSubmit = async (values: IUser) => {
        await signup(values);
    }   
    
  return (
    <DynamicForm initialValues={initialUserValues} schema={signupSchema} handleSubmit={handleSubmit} />
  )
}

export default SignupForm