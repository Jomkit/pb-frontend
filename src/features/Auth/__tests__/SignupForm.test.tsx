import { render } from "@testing-library/react";
import SignupForm from "../SignupForm";

describe("signup form", function(){
    const mockSignup = vi.fn();

    it('should render without crashing', () => { 
        render(<SignupForm signup={mockSignup} />);
    })

    it('should match snapshot', () => { 
        const { asFragment } = render(
            <SignupForm signup={mockSignup} />
        )

        expect(asFragment()).toMatchSnapshot();
    })
})