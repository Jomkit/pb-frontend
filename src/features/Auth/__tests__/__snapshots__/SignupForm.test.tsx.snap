// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`signup form > should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex flex-col text-start w-1/2 m-auto"
      data-testid="flowbite-card"
    >
      <div
        class="flex h-full flex-col justify-center gap-4 p-6"
      >
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="username"
        >
          username
        </label>
        <input
          class="form-field1 focus:outline-none focus:shadow-outline"
          id="username"
          name="username"
          type="text"
          value=""
        />
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="password"
        >
          password
        </label>
        <input
          class="form-field1 focus:outline-none focus:shadow-outline"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="firstName"
        >
          firstName
        </label>
        <input
          class="form-field1 focus:outline-none focus:shadow-outline"
          id="firstName"
          name="firstName"
          type="text"
          value=""
        />
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="lastName"
        >
          lastName
        </label>
        <input
          class="form-field1 focus:outline-none focus:shadow-outline"
          id="lastName"
          name="lastName"
          type="text"
          value=""
        />
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="email"
        >
          email
        </label>
        <input
          class="form-field1 focus:outline-none focus:shadow-outline"
          id="email"
          name="email"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      class="btn bg-blue-700 text-white"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
